ibis/backends/druid/__init__.py

```diff
@@ -44,35 +44,20 @@ def version(self) -> str:
             [(version,)] = result.fetchall()
         return version
 
-    def _from_url(self, url: ParseResult, **kwargs):
-        """Connect to a backend using a URL `url`.
-
-        Parameters
-        ----------
-        url
-            URL with which to connect to a backend.
-        kwargs
-            Additional keyword arguments
-
-        Returns
-        -------
-        BaseBackend
-            A backend instance
-
-        """
-        kwargs = {
-            "user": url.username,
-            "password": unquote_plus(url.password)
-            if url.password is not None
-            else None,
-            "host": url.hostname,
-            "path": url.path,
-            "port": url.port,
-            **kwargs,
-        }
-
+    def _from_url(self, url: ParseResult, **kwarg_overrides):
+        kwargs = {}
+        if url.username:
+            kwargs["user"] = url.username
+        if url.password:
+            kwargs["password"] = unquote_plus(url.password)
+        if url.hostname:
+            kwargs["host"] = url.hostname
+        if url.path:
+            kwargs["path"] = url.path
```

Comment:
this would be a followup PR, but do we want to make the backends follow a consistent format for their kwargs, ie for Druid should we rename the `path` kwarg of do_connect() to `database` to be consistent with the other backends? bigquery is strange, with its project_id and dataset_id. Should we rename those to catalog and database? IDK, at what point do we switch from using the backend specific terminology to using the generic ibis terms of catalog and database? I could see that being during backend construction time you use the backend-specific terminology, but once a backend is instantiated all methods should use the same API with the ibis terms of catalog and database.

Reply:
> I could see that being during backend construction time you use the backend-specific terminology, but once a backend is instantiated all methods should use the same API with the ibis terms of catalog and database.

Yeah, we've settled on this in the past, because the backends might take radically different kwargs, and it makes it simpler for someone to take a working connection using a given DB python dbapi client and then use the "same" connection string with Ibis without having to translate all the argument names.

Reply:
Sounds good. I added this explanation in commit [`8bcb85b` (#11112)](https://github.com/ibis-project/ibis/pull/11112/commits/8bcb85b84d10c60d8770888d6329e4b9827c6e05) as part of PR https://github.com/ibis-project/ibis/pull/11112